From ce05123e7a35cc1c80b464feb51427353ff79e0a Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Tue, 18 Apr 2017 13:50:15 -0400
Subject: [PATCH 33/43] hsr/prp: fix debugfs so that multiple instances can be
 supported

Use the netdev name to create the debugfs file for hsr and prp so that
multiple instances can be supported.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 net/hsr-prp/hsr_prp_debugfs.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/net/hsr-prp/hsr_prp_debugfs.c b/net/hsr-prp/hsr_prp_debugfs.c
index 62deae3..7cf2cac 100644
--- a/net/hsr-prp/hsr_prp_debugfs.c
+++ b/net/hsr-prp/hsr_prp_debugfs.c
@@ -149,11 +149,7 @@ int hsr_prp_debugfs_init(struct hsr_prp_priv *priv,
 	int rc = -1;
 	struct dentry *de = NULL;
 
-	if (priv->prot_version <= HSR_V1)
-		de = debugfs_create_dir("hsr", NULL);
-	else
-		de = debugfs_create_dir("prp", NULL);
-
+	de = debugfs_create_dir(hsr_prp_dev->name, NULL);
 	if (!de) {
 		netdev_err(hsr_prp_dev, "Cannot create hsr-prp debugfs root\n");
 		return rc;
-- 
1.9.1

